app.js

```diff
@@ -219,28 +222,58 @@ app.post('/interactions', verifyKeyMiddleware(process.env.PUBLIC_KEY), async (re
             };
 
             const lbargs = args[0].split('+');
-            if (lbargs[1] ==='totd') {
-                track_info.endTimestamp = Number(convertNumberToBase(lbargs[2], 64, 10));
-            }
-            if (lbargs[lbargs.length - 1] === 'f') {
-                args.push(0);
-            } else if (lbargs[lbargs.length - 1] === 'l') {
-                args.push(1000-args[3]);
-            } else if (lbargs[lbargs.length - 1] === 'p') {
-                data.values[0].split(';').forEach((e) => {
-                    args.push(e);
-                });
-            }
+            if (lbargs[1] === 'totd') track_info.endTimestamp = Number(convertNumberToBase(lbargs[2], 64, 10));
+            let body;
+            if (lbargs[lbargs.length - 1] ==='i') {
+                let leaderboard = await trackmaniaFacade.getLeaderboardInfo(track_info, args[3], true, args[4]);
```

Comment:
Similar comment as above for `args`. Not sure if the structure is fixed enough for us to have one single unpacking method, but extracting these to variables would make it slightly easier to read
```suggestion
                const length = args[3];
                const offset = args[4];
                let leaderboard = await trackmaniaFacade.getLeaderboardInfo(track_info, length, true, args[4]);
```

This would especially help in the code slightly below where don't have the context of method parameter names to help us (or in code review where I don't immediately have the parameter names xd)
```js
...
                } else if (lbargs[lbargs.length - 1] === 'l') {
                    args.push(1000-args[3]);
                }
...
```

Reply:
new commit i think i did what you asked